### PR TITLE
Add direct tpm support for spire-agent

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

File: charts/spire/charts/spire-agent/templates/configmap.yaml
Comment (kfox1111) [inline charts/spire/charts/spire-agent/templates/configmap.yaml:72]:
```suggestion
      plugin_checksum: {{ .plugin.checksum }}
```

File: charts/spire/charts/spire-agent/templates/configmap.yaml
Comment (kfox1111) [inline charts/spire/charts/spire-agent/templates/configmap.yaml:59]:
```suggestion
{{- $nodeAttestorUsed := add (len nodeAttestor) (len .Values.unsupportedBuiltInPlugins.nodeAttestor) }}
```

File: examples/tpm-direct/values.yaml
Comment (kfox1111) [inline examples/tpm-direct/values.yaml:13]:
```suggestion
  nodeAttestor:
    tpmDirect:
```

File: examples/tpm-direct/values.yaml
Comment (marcofranssen) [inline examples/tpm-direct/values.yaml:12]: Should the chart check this `restriction.kubernetes.io` is part of the template as some sort of safetynet?

File: charts/spire/charts/spire-agent/values.yaml
Comment (marcofranssen) [inline charts/spire/charts/spire-agent/values.yaml:169]: Maybe just name this. Trying to understand what Direct means in this context.

```suggestion
  tpm:
```

**Review by marcofranssen — COMMENTED** [review]
Small question and suggestion. Looks pretty neat.

File: charts/spire/charts/spire-agent/values.yaml
Comment (kfox1111) [reply charts/spire/charts/spire-agent/values.yaml:169]: Marked as Direct to keep room for a second feature in the future. There is a second tpm mode that spire supports here: https://github.com/spiffe/spire/blob/main/doc/plugin_agent_nodeattestor_tpm_devid.md

Direct uses the tpm's private key hash directly for identification. This is the simplest way to use a tpm.

DeviceID requires an extra server and an out of band registration mechanism. But is the only way to attest some devices (Apple I think)

File: examples/tpm-direct/values.yaml
Comment (kfox1111) [reply examples/tpm-direct/values.yaml:12]: I wish we could, but it is a restriction on the labels being manually added to the node by the end user. Its not done in the chart, so not something we can check in the chart.

k8s itself does enforce the restriction.If you try and load it all in one string, it complains that the label is too long.

Once the tpm plugin pr's merge, I'm going to provide documentation at https://github.com/spiffe/spiffe.io/pull/290

File: charts/spire/charts/spire-agent/values.yaml
Comment (marcofranssen) [reply charts/spire/charts/spire-agent/values.yaml:169]: ok, thanks for explaining, makes sense to keep room for extension. Is `direct` a uniformal way of referring to this tpm approach? If so I'm fine. Just wanter to pick your brain on the naming convention.

File: examples/tpm-direct/values.yaml
Comment (marcofranssen) [reply examples/tpm-direct/values.yaml:12]: Yes outside of the chart an enduser could do that indeed. But this value specifically we can control it right?

File: charts/spire/charts/spire-agent/values.yaml
Comment (kfox1111) [reply charts/spire/charts/spire-agent/values.yaml:169]: I don't know of any standard naming around it. Also open to different names. Though naming things is hard.

File: examples/tpm-direct/values.yaml
Comment (kfox1111) [reply examples/tpm-direct/values.yaml:12]: Oh. I think I may have been misunderstanding what you've been saying...  I thought you were talking about the character limit, not the parentIDTemplate itself.

That would be good to do if we could figure out a way to do it. I can't think of a way off hand to be able to do that, outside of writing our own gotl parser?

Comment (faisal-memon) [reply examples/tpm-direct/values.yaml:12]: @marcofranssen Are you asking just to check if the string `restriction.kubernetes.io` is present?

File: charts/spire/charts/spire-agent/values.yaml
Comment (marcofranssen) [reply charts/spire/charts/spire-agent/values.yaml:171]: Yup agree. I'm not very into tpm stuff so just asked my 101 questions here on the PR. I guess it is good to go. @faisal-memon feel free to approve and merge if you don't see any other hurdles.

File: examples/tpm-direct/values.yaml
Comment (marcofranssen) [reply examples/tpm-direct/values.yaml:12]: it was about the char limit indeed, but seeing Kevins response that is not possible prior to parsing the template I guess. So no worries, I'm fine merging like this. Just tried to see if we could failsafe the user.

File: charts/spire/charts/spire-agent/templates/daemonset.yaml
Comment (faisal-memon) [inline charts/spire/charts/spire-agent/templates/daemonset.yaml:65]: Can you add a comment here explaining what this snippet does?

File: charts/spire/charts/spire-agent/templates/daemonset.yaml
Comment (kfox1111) [inline charts/spire/charts/spire-agent/templates/daemonset.yaml:59]:
```suggestion
              # Fetch the TPM's fingerprint from the TPM and output it on the Computers Console. This enables secure bootstrapping of the Node with the Server.
              TPM=`/app/get_tpm_pubhash 2> /dev/null`
```

**Review by faisal-memon — APPROVED** [review]

File: charts/spire/charts/spire-agent/values.yaml
Comment (faisal-memon) [reply charts/spire/charts/spire-agent/values.yaml:174]: Naming looks ok to me. Keeping room for the second tpm plugin makes sense

**Pull request merged**
